code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -1088,40 +1186,53 @@ func (sc *StorageSmartContract) updateAllocationRequestInternal(
 		}
 	}
 
-	cp, err := sc.getChallengePool(alloc.ID, balances)
-	if err != nil {
-		return "", common.NewError("allocation_updating_failed", err.Error())
+	var cpBalance currency.Coin
+	if !isEnterprise {
+		cp, err := sc.getChallengePool(alloc.ID, balances)
+		if err != nil {
+			return "", common.NewError("allocation_updating_failed", err.Error())
+		}
+
+		cpBalance = cp.Balance
 	}
 
-	tokensRequiredToLock, err := alloc.requiredTokensForUpdateAllocation(cp.Balance, request.Extend, t.CreationDate)
+	tokensRequiredToLock, err := alloc.requiredTokensForUpdateAllocation(cpBalance, request.Extend, t.CreationDate)
 	if err != nil {
 		return "", common.NewError("allocation_updating_failed", err.Error())
 	}
 
-	if t.Value < tokensRequiredToLock {
-		actErr := chainstate.WithActivation(balances, "demeter", func() error {
+	actErr = chainstate.WithActivation(balances, "electra", func() error {
+		if t.Value < tokensRequiredToLock {
 			return common.NewError("allocation_updating_failed",
-				fmt.Sprintf("not enough tokens to cover update allocation cost (locked : %d < required : %d)", t.Value, tokensRequiredToLock))
-		}, func() error {
+				fmt.Sprintf("not enough tokens to cover update allocation cost (locked : %d < required : %d)", t.Value, tokensRequiredToLock+t.Value))
+		}
+		return nil
+	}, func() error {
+		if tokensRequiredToLock > 0 {
```

Comment:
Should be `if tokensRequiredToLock > 0 && t.Value < tokenRequiredToLock`, otherwise as long as the `tokensRequiredToLock` is greater than 0, we will see error.

Reply:
We are adding value to balance in the very starting. So if there are any extra tokens required then we need to return the error.